packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMainInput.tsx

```diff
@@ -86,6 +87,8 @@ function SourceChainTokenBalance({
   const [networks] = useNetworks()
   const { isDepositMode, childChainProvider } =
     useNetworksRelationship(networks)
+  const nativeCurrencyDecimalsOnSourceChain =
+    useNativeCurrencyDecimalsOnSourceChain()
 
   const nativeCurrencyBalances = useNativeCurrencyBalances()
   const selectedTokenBalances = useSelectedTokenBalances()
```

Comment:
you have to do a sweep of `nativeCurrency.decimals` in the codebase and try to replace them with `useNativeCurrencyDecimalsOnSourceChain`'s value because for example `useSelectedTokenBalances()` uses `nativeCurrency.decimals`, and if you type in the amount input field for L3 withdrawals, the decimal truncation would be wrong